web-ui/src/main/resources/catalog/components/solr/SolrRequest.js

```diff
@@ -0,0 +1,699 @@
+/*
+ * Copyright (C) 2001-2016 Food and Agriculture Organization of the
+ * United Nations (FAO-UN), United Nations World Food Programme (WFP)
+ * and United Nations Environment Programme (UNEP)
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or (at
+ * your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301, USA
+ *
+ * Contact: Jeroen Ticheler - FAO - Viale delle Terme di Caracalla 2,
+ * Rome - Italy. email: geonetwork@osgeo.org
+ */
+
+(function() {
+  goog.provide('gn_solr_request');
+
+  var module = angular.module('gn_solr_request', []);
+
+  var solrRequestEvents = {
+    search: 'search'
+  };
+
+  var ROWS = 20;
+  var FACET_RANGE_COUNT = 5;
+  var FACET_RANGE_DELIMITER = ' - ';
+
+  geonetwork.GnSolrRequest = function(config, $injector) {
+    this.$http = $injector.get('$http');
+    this.$q = $injector.get('$q');
+    this.$translate = $injector.get('$translate');
+    this.urlUtils = $injector.get('gnUrlUtils');
```

Comment:
I guess is due to my ignorance of some AngularJs stuff, but whats the diff between `$injector.get('$http')` and sending `$http` as a parameter of the function (same for the other `$injector.get` calls.